package.json

```diff
@@ -43,9 +44,20 @@
     "@types/jest": "^29.5.3",
     "eslint": "^8.46.0",
     "eslint-config-prettier": "^8.9.0",
+    "eslint-plugin-jest": "^27.2.3",
+    "husky": "^8.0.3",
+    "lint-staged": "^14.0.0",
+    "mongodb-memory-server": "^8.15.1",
     "nodemon": "^3.0.1",
     "prettier": "^3.0.0",
+    "supertest": "^6.3.3",
     "swagger-autogen": "^2.23.5",
     "swagger-jsdoc": "^6.2.8"
+  },
+  "lint-staged": {
+    "*.js": [
+      "npm run lint:fix",
+      "npm run format"
+    ]
```

Comment:
@NatachaKey this is similar to what is set up in the frontend repo. It will lint and format code automatically for you whenever you do `git commit`.